homeassistant/components/smappee_official/config_flow.py

```diff
@@ -0,0 +1,30 @@
+"""Config flow for Smappee Official."""
+import logging
+
+from homeassistant import config_entries
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SmappeeFlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle Smappee Official OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason="single_instance_allowed")
```

Comment:
We need to test this case too.

Add a mock config entry to hass and then start the flow and assert that the flow aborts.

https://github.com/home-assistant/core/blob/9a867cbb7568074e21193fd2e295c3ba95c9cedb/tests/common.py#L668-L706